velox/exec/OutputBuffer.cpp

```diff
@@ -695,7 +695,7 @@ double OutputBuffer::getUtilization() const {
 }
 
 bool OutputBuffer::isOverutilized() const {
-  return (totalSize_ > (0.5 * maxSize_)) && !atEnd_;
+  return (totalSize_ > (0.5 * maxSize_)) || atEnd_;
```

Comment:
This change is a bit strange. Why would we report the buffer being over-utilized when it is no longer used at all? If there is some weird behavior on Presto coordinator that "requires" this, then perhaps a better place for that would be in Prestissimo where we could report is-over-utilized by checking both OutputBuffer::isOverutilized and OutputBuffer::isAtEnd.

CC: @xiaoxmeng @kewang1024

Reply:
If we keep this change, then let's update the comment:

```
  // Indicates if this output buffer is over-utilized, i.e. at least half full,
  // and will start blocking producers soon. This is used to dynamically scale
  // the number of consumers, for example, increase number of TableWriter tasks.
  bool isOverutilized() const;
```